server/app/controllers/admin/AdminReportingController.java

```diff
@@ -57,31 +57,40 @@ private CiviFormProfile getCiviFormProfile(Http.Request request) {
 
   @Secure(authorizers = Authorizers.Labels.ANY_ADMIN)
   public Result show(Http.Request request, String programSlug) {
-    String programName =
-        programService
-            .getActiveFullProgramDefinitionAsync(programSlug)
-            .toCompletableFuture()
+    String programLocalizedName =
+        programRepository
```

Comment:
**suggestion:** To avoid calling getProgramDefinition, we should change this to:

`programRepository.getShallowProgramDefinition(programRepository.getActiveProgramFromSlug(programSlug).join()).localizedName().getDefault()`

Reply:
I don't mind using either. should I change it back to programService? I don't know if programService has shallow Program Definition